tests/phpunit/vfsStreamWrapperErroneousFileTestCase.php

```diff
@@ -0,0 +1,364 @@
+                    8 => 0,
+                    9 => 0,
+                    10 => 0,
+                    11 => -1,
+                    12 => -1,
+                    'dev' => 0,
+                    'ino' => spl_object_id($file),
+                    'mode' => 0100666,
+                    'nlink' => 0,
+                    'uid' => vfsStream::getCurrentUser(),
+                    'gid' => vfsStream::getCurrentGroup(),
+                    'rdev' => 0,
+                    'size' => 0,
+                    'atime' => 0,
+                    'mtime' => 0,
+                    'ctime' => 0,
+                    'blksize' => -1,
+                    'blocks' => -1,
+                ]
+            )
+        );
+    }
+
+    public function testLockWithErrorMessageTriggersError(): void
+    {
+        $message = uniqid();
+        $file = vfsStream::newErroneousFile('foo', ['lock' => $message])->at($this->root);
+
+        expect(static function () use ($file): void {
+            $fh = fopen($file->url(), 'r');
+            flock($fh, LOCK_SH);
+        })->triggers(E_USER_WARNING)->withMessage($message);
+    }
+
+    public function testLockWithErrorMessageReturnsFalse(): void
+    {
+        $file = vfsStream::newErroneousFile('foo', ['lock' => uniqid()])->at($this->root);
+
+        $fh = @fopen($file->url(), 'r');
+        $actual = @flock($fh, LOCK_SH);
+        @fclose($fh); // Close calls lock to unlock
+
+        assertFalse($actual);
+    }
+
+    public function testFilemtimeWithErrorMessageTriggersError(): void
+    {
+        $message = uniqid();
+        $file = vfsStream::newErroneousFile('foo', ['stat' => $message])->at($this->root);
+
+        expect(static function () use ($file): void {
+            filemtime($file->url());
+        })->triggers(E_USER_WARNING)->withMessage($message);
+    }
+
+    public function testFilemtimeWithErrorMessageReturnsZero(): void
+    {
+        $file = vfsStream::newErroneousFile('foo', ['stat' => uniqid()])->at($this->root);
+
+        $actual = @filemtime($file->url());
+
+        assertThat($actual, equals(0));
+    }
+
+    public function testFilemtimeWithoutError(): void
+    {
+        $file = vfsStream::newErroneousFile('foo', [])->at($this->root);
+
+        $actual = filemtime($file->url());
+
+        assertThat($actual, equals(time(), 1));
+    }
+
+    public function testFileatimeWithErrorMessageTriggersError(): void
+    {
+        $message = uniqid();
+        $file = vfsStream::newErroneousFile('foo', ['stat' => $message])->at($this->root);
+
+        expect(static function () use ($file): void {
+            fileatime($file->url());
+        })->triggers(E_USER_WARNING)->withMessage($message);
+    }
+
+    public function testFileatimeWithErrorMessageReturnsZero(): void
+    {
+        $file = vfsStream::newErroneousFile('foo', ['stat' => uniqid()])->at($this->root);
+
+        $actual = @fileatime($file->url());
+
+        assertThat($actual, equals(0));
```

Comment:
According to the manual all `fileXtime()` functions should return `false` on failure. To achieve this my guess is that `vfsStreamWrapper::stream_stat()` must act accordingly when any of the functions of `vfsStreamErroneousFile` called by it trigger an error.

Reply:
Fixed